presto-main/src/main/java/com/facebook/presto/server/PrestoServer.java

```diff
@@ -131,6 +131,24 @@ public void run()
             injector.getInstance(Announcer.class).start();
 
             log.info("======== SERVER STARTED ========");
+
+            ServerConfig serverConfig = injector.getInstance(ServerConfig.class);
+
+            if (serverConfig.isCoordinator() && serverConfig.isEnabledUIonSecondaryPort()) {
+                Bootstrap uIApp = new Bootstrap(new NodeModule(),
+                                                new DiscoveryModule(),
+                                                new JsonModule(),
+                                                new JaxrsModule(true),
+                                                new JmxModule(),
+                                                new JmxHttpModule(),
+                                                new JsonEventModule(),
+                                                new HttpEventModule(),
+                                                new ServerSecurityModule(),
+                                                new CoordinatorUIHttpServerModule(injector));
+                Injector uIInjector = uIApp.doNotInitializeLogging().initialize();
+
+                log.info("======== SECONDARY UI STARTED ========");
```

Comment:
When the server is started, the logging should show "======== SERVER STARTED ========". No need for "======== SECONDARY UI STARTED ========" IMO. I think it would be best to not introduce a new "secondary UI concept" and just have the concept of a UI port and a protocol (or something like that) port.